src/tiled/changemapobject.cpp

```diff
@@ -99,35 +99,39 @@ ChangeMapObjectsTile::ChangeMapObjectsTile(MapDocument *mapDocument,
     , mMapObjects(mapObjects)
     , mTile(tile)
 {
+    mCell = Cell(mTile);
     for (MapObject *object : mMapObjects) {
-        Tile *tile = object->cell().tile();
-        mOldTiles.append(tile);
-        mSizePreserved.append(object->size() == tile->size());
+        Cell cell = object->cell();
+        mOldCells.append(cell);
+        mOldSizes.append(object->size());
+        Tile *tile = cell.tile();
+        mUpdateSize.append(tile == nullptr || object->size() == tile->size());
```

Comment:
Thank you, my initial implementation left the object size as is, but I wasn't sure so I just updated the size to match the tile, should've discussed that.